WcaOnRails/app/models/competition.rb

```diff
@@ -1783,7 +1784,9 @@ def update_persons_wcif!(wcif_persons, current_user)
       # NOTE: person doesn't necessarily have corresponding registration (e.g. registratinless organizer/delegate).
       if wcif_person["roles"]
         roles = wcif_person["roles"] - ["delegate", "trainee-delegate", "organizer"] # These three are added on the fly.
-        registration.update!(roles: roles)
+        # The additional roles are only for WCIF purposes and we don't validate them,
+        # so we can safely skip validations by using update_attribute
+        registration.update_attribute(:roles, roles)
```

Comment:
Validating a registration involves a couple additional queries. We could avoid some of them via preload, but if the competition has qualifications we make a query checking against competitor's prior results, which we can't preload reasonably. In this case we only want to update the roles and can skip the validations altogether.